gio/src/list_store.rs

```diff
@@ -75,3 +76,23 @@ unsafe extern "C" fn compare_func_trampoline(
 
     (*func)(&a, &b).into_glib()
 }
+
+impl<A: IsA<glib::Object>> std::iter::Extend<A> for ListStore {
+    fn extend<T: IntoIterator<Item = A>>(&mut self, iter: T) {
+        unsafe {
+            let additions = iter
+                .into_iter()
+                .map(|o| o.as_ptr() as *mut glib::gobject_ffi::GObject)
+                .collect::<Vec<_>>();
+            let n_additions = additions.len() as u32;
+
+            ffi::g_list_store_splice(
```

Comment:
Why via FFI instead of https://gtk-rs.org/gtk-rs-core/git/docs/gio/struct.ListStore.html#method.splice?

Reply:
Because splice takes `&[A]`, then collects the items in a `Vec`.
To call `splice` with a `&[A]` I have to collect the iterator's items in a `Vec`.

Calling splice would allocate a `Vec` two times.

Reply:
Yes that's needed actually :) See https://github.com/gtk-rs/gtk-rs-core/pull/421#discussion_r770784002

Reply:
Or rather: you can collect all objects into a `Vec<glib::Object>` and then pass it to the FFI function if you want, without another vec. `your_vec.as_ptr() as *const _ as *const *const glib::gobject_ffi::GObject` is valid.

If that's also valid to do for `splice()`, not sure. I'd have to think about that a bit (please create an issue).

Reply:
Yup, I'm now convinced I need to allocate 2 times anyway